File: src/main/java/com/refinedmods/refinedstorage/apiimpl/autocrafting/CraftingPatternFactoryException.java
package com.refinedmods.refinedstorage.apiimpl.autocrafting;

import net.minecraft.network.chat.Component;

public class CraftingPatternFactoryException extends Exception {
    private final transient Component errorMessage;

    public CraftingPatternFactoryException(Component errorMessage) {
        this.errorMessage = errorMessage;
    }

    public Component getErrorMessage() {
        return errorMessage;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/apiimpl/autocrafting/CraftingPatternFactoryException.kt
package com.refinedmods.refinedstorage.apiimpl.autocrafting

import net.minecraft.util.text.Text


class CraftingPatternFactoryException(errorMessage: Text) : Exception() {
    private val errorMessage: Text
    fun getErrorMessage(): Text {
        return errorMessage
    }

    init {
        this.errorMessage = errorMessage
    }
}